Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs

```diff
@@ -811,8 +811,14 @@ private Participant AddNewParticipantToContest(Contest contest, bool official)
                             Resource.ContestsGeneral.User_is_not_registered_for_exam);
                     }
                 }
-                catch
+                catch (Exception ex)
                 {
+                    var httpEx = ex as HttpException;
+                    if (httpEx?.GetHttpCode() == (int)HttpStatusCode.Forbidden)
+                    {
+                        throw;
```

Comment:
Can we throw our own specific error here in order to prevent the user from seeing real exception messages, just in case the HttpClient gets a real Forbidden exception.